libp2p/host/basic_host.py

```diff
@@ -41,13 +45,18 @@ class BasicHost(IHost):
     multiselect: Multiselect
     multiselect_client: MultiselectClient
 
-    def __init__(self, network: INetwork, router: KadmeliaPeerRouter = None) -> None:
+    def __init__(
+        self,
+        network: INetwork,
+        router: KadmeliaPeerRouter = None,
+        default_protocols: "OrderedDict[TProtocol, StreamHandlerFn]" = DEFAULT_HOST_PROTOCOLS,
```

Comment:
Sorry, I found I missed this one when I reviewed last time. `DEFAULT_HOST_PROTOCOLS` seems fine to be shared across `BasicHost` instances since we are probably not going to modify it, but I'm a little bit worried `Multiselect` might accidentally modify it. IMO we have two options here: 1) still initialize `default_protocols` in the function body. 2) change `Multiselect`, make it not modify the passed `handlers`(i.e. we probably need to remove `add_handler`).